### ISSUE TITLE
Improve loc / error message for auto-imported libs

### ISSUE THREAD

**Issue by sorawee** [brownplt/pyret-lang#678]
```
a = 1
list = 1
```

when running, i get the following error:

```
It looks like you've defined the name list twice, at
definitions: line 1, column 0 and
definitions: line 2, column 0.
You need to pick a different name for one of them.
```

where `definitions: line 1, column 0` spans the whole code (from line 1 to line 2), and `definitions: line 2, column 0` spans only over the word `list`.

First, I didn't _defined the name list twice_. I did it only once. Second, the location of `definitions: line 1, column 0` is just weird.

Would be nice to have an error message like `raw-array`:

```
The name raw-array is already defined. You need to pick a different name for raw-array at definitions: xxx
```

though I understand that this might be difficult because built-in functions are not the same as functions from libs.

**Comment by blerner** [comment]
The problem appears to be in resolve-scope, and more generally with how modules export values.  I haven't traced all the details yet, but basically we assign "the entire module" as the srcloc of ambient values.  And fixing it is tricky, because our exported information doesn't provide srclocs of where values came from.  (This will come up again, I suspect, when handling include statements...)

Note that the message you see is in fact the same error structure for both list and raw-array.  It's just that the rendering of it checks whether the initial srcloc is a builtin or not, and chooses which message to render as a result.  Which means once we get builtin locations for the ambient values, this will probably go away.

**Comment by blerner** [comment]
The commit mentioned above fixes the initial example, but the following program still fails badly:
```
include world

big-bang = 5
```
gives
```
It looks like you've defined the name big-bang twice, at
definitions: line 1, column 0 and
definitions: line 3, column 0.
You need to pick a different name for one of them.
```
where the first location spans the entire document.  I wonder whether we shouldn't have a third kind of srcloc, an `included`, that highlights the location of the `include` line that brought the name into scope.  Neither `srcloc` nor `builtin` quite have the right implication here: if I include a file that I've written, then a duplicate name is neither builtin nor defined twice in this file.  @jpolitz thoughts?

**Comment by sorawee** [comment]
Related:

```
a = 1
lists = 3
```

fails like the original program posted above.

**Comment by schanzer** [comment]
@sorawee and @blerner The error is a lot better now. OK to close?

**Comment by blerner** [comment]
No.  The world/big-bang error still occurs on #horizon, and that's a problem.

**Comment by schanzer** [comment]
@blerner the world/big-bang error seems much improved:
![image](https://user-images.githubusercontent.com/77126/90049225-81e06180-dca2-11ea-81c2-b9cfef52f141.png)


OK to close?
